packages/@aws-cdk/toolkit-lib/lib/toolkit/toolkit.ts

```diff
@@ -249,6 +252,145 @@ export class Toolkit extends CloudAssemblySourceBuilder implements AsyncDisposab
     return new IdentityCloudAssemblySource(assembly.assembly);
   }
 
+  /**
+   * Diff Action
+   */
+  public async diff(cx: ICloudAssemblySource, options: DiffOptions): Promise<number> {
```

Comment:
```suggestion
  public async diff(cx: ICloudAssemblySource, options: DiffOptions): Promise<TemplateDiff> {
```

Reply:
ack. will do before this becomes a reviewable PR